### PR TITLE
Cmssw 80 x sherpa weights

### PR DESCRIPTION
**Pull request by thuer**
Backport of the changes in 81X (#13756) 

Adding the possibility of additional weights (scale / pdf variation) for Sherpa 2.2. Sherpa stores the weights in form of named weights which we do not do in CMSSW, so the relevant names are stored in GenLumiInfoHeader for further use.

### CODE REVIEW

**Comment by covarell** [comment]
+1

File: GeneratorInterface/SherpaInterface/src/SherpaHadronizer.cc [hunk 1]
```diff
@@ -46,15 +46,17 @@ class SherpaHadronizer : public gen::BaseHadronizer {
   void statistics();
   bool generatePartonsAndHadronize();
   bool decay();
+  bool rearrangeWeights;
```
Comment (davidlange6) [inline GeneratorInterface/SherpaInterface/src/SherpaHadronizer.cc]: @thuer - why make this variable public?

Comment (thuer) [inline GeneratorInterface/SherpaInterface/src/SherpaHadronizer.cc]: for no reason... made it private

**Comment by covarell** [comment]
+1

**Comment by vciulli** [comment]
what is holding on this PR? 
would it be possible to merge it and request a new release at the next ORP? 
it is needed for EXOTICA mc production 
https://hypernews.cern.ch/HyperNews/CMS/get/generators/2997/1.html

**Comment by davidlange6** [comment]
@vciulli - frequent lack of gen reps at release meetings... what is the connection between 80x and MC production?

**Comment by vciulli** [comment]
@davidlange6 - 80x is used for GEN in Fastsim MC production.
However there aren't urgent requests for Sherpa in this release (while there are some in 71x), so this PR can be just scheduled for the next available release. Hope this clarify GEN request.
Sorry if I couldn't be present at the ORP because of another meeting I had at the same time.

**Comment by vciulli** [comment]
As we will move soon to 81X for MC production, this PR is no more needed and can be closed. 
Thanks.

**Pull request closed without merging**